Login fix and Logout

benasbart opened the pull request.

nicki419 reviewed (changes requested):
Works well and does what it should. Would be nice to see a sort of loading animation on the login page though as soon as the user hits enter or clicks on login, so they know something is happening.

benasbart commented:
Properly fixed the redirection to dashboard with a login state service.
Switched to mudblazor components in login page and added animation to button.

nicki419 reviewed (approved):
Beautiful. Well done!
